packages/multichain/src/scope/types.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  isCaipNamespace,
+  isCaipChainId,
+  parseCaipChainId,
+} from '@metamask/utils';
+import type {
+  CaipChainId,
+  CaipReference,
+  CaipAccountId,
+  KnownCaipNamespace,
+  CaipNamespace,
+  Json,
+} from '@metamask/utils';
+
+// These External prefixed types represent the CAIP-217
+// Scope and ScopeObject as defined in the spec.
+export type ExternalScopeString = CaipChainId | CaipNamespace;
+export type ExternalScopeObject = Omit<ScopeObject, 'accounts'> & {
+  references?: CaipReference[];
+  accounts?: CaipAccountId[];
+};
+export type ExternalScopesObject = Record<
+  ExternalScopeString,
+  ExternalScopeObject
+>;
```

Comment:
These seem like important types. Is it worth using JSDoc here to capture this note?

``` suggestion
/**
 * Represents a `scopeString` as defined in [CAIP-217](https://chainagnostic.org/CAIPs/caip-217).
 */
export type ExternalScopeString = CaipChainId | CaipNamespace;

/**
 * Represents a `scopeObject` as defined in [CAIP-217](https://chainagnostic.org/CAIPs/caip-217).
 */
export type ExternalScopeObject = Omit<ScopeObject, 'accounts'> & {
  references?: CaipReference[];
  accounts?: CaipAccountId[];
};

/**
 * Represents a keyed `scopeObject` as defined in [CAIP-217](https://chainagnostic.org/CAIPs/caip-217).
 */
export type ExternalScopesObject = Record<
  ExternalScopeString,
  ExternalScopeObject
>;

Reply:
done here: https://github.com/MetaMask/core/pull/4784/commits/3f56531543e1b1d07729e78bc4b70d40d8369df5

Added a todo comment to push update to CAIP-217 per my comment [here](https://github.com/MetaMask/core/pull/4784#discussion_r1833134198)